### PR TITLE
add the NewPidFileFn to helper

### PR DESCRIPTION
**Pull request by sbookworm**
for the enhancement: https://github.com/prometheus/client_golang/issues/760

### CODE REVIEW

**Review by beorn7 — CHANGES_REQUESTED** [review]
Thank you for your contribution. It looks good in general. Just minor comments below.

Another thing is the file naming. `helper.go` suggest we will put more helper function in there later. I would either call the file `pid_file_helper.go` (and the test file accordingly), or perhaps even better, just add it to `process_collector.go`, because it's not that much new code, and it is closely tied to the code in `process_collector.go`.

File: prometheus/helper_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+// TestNewPidFileFn ...
+func TestNewPidFileFn(t *testing.T) {
+	folderPath, err := os.Getwd()
+	if err != nil {
+		t.Error("failed to get current path")
+	}
+	mockPidFilePath := filepath.Join(folderPath, "mockPidFile")
+	defer cleanMockPidFile(mockPidFilePath)
```
[the remark this answers is not in the document]
Comment (sbookworm) [reply prometheus/helper_test.go:42]: good idea :+1:

**Comment by sbookworm** [comment]
@beorn7  sorry for using the force option when submitting the changes.
There is only one latest commit now :(  please help to review, thanks

**Review by beorn7 — CHANGES_REQUESTED** [review]
Code looks good now. But you still have it in files called `helper.go` and `helper_test.go`, which suggests those are files for helper code. But there are no plans right now to add more "helper" code to them. My suggestion is to move the code from `helper.go` to `process_collector.go` and the code from `helper_test.go` to `process_collector_test.go`.

**Comment by beorn7** [comment]
> sorry for using the force option when submitting the changes

That's fine for your own branch if it is unlikely that those changes are used somewhere else already.

File: prometheus/process_collector_test.go [hunk 2]
```diff
@@ -101,3 +104,65 @@ func TestProcessCollector(t *testing.T) {
 		t.Errorf("%d metrics collected, want 1", n)
 	}
 }
+
+// TestNewPidFileFn ...
```
Comment (beorn7) [inline prometheus/process_collector_test.go:108]: Final nit: Test functions don't require a doc comment. If you want to provide a meaningful doc comment, go for it. But this one is not meaningful. I'd just delete this line.

**Review by beorn7 — CHANGES_REQUESTED** [review]
Thank you. Just one final thing to do.

**Review by beorn7 — APPROVED** [review]
Thank you.

**Pull request merged**